lib/Conversion/TorchOnnxToTorch/DefaultDomainAtoF.cpp

```diff
@@ -941,6 +941,144 @@ void mlir::torch::onnx_c::populateDefaultDomainAtoF(
+            binder.s64IntegerArrayAttr(dilations, "dilations",
+                                       SmallVector<int64_t>{1, 1}) ||
+            binder.s64IntegerArrayAttr(strides, "strides",
+                                       SmallVector<int64_t>{1, 1}) ||
+            binder.s64IntegerArrayAttr(pads, "pads",
+                                       SmallVector<int64_t>{0, 0, 0, 0}))
+          return failure();
+
+        auto imageShapeTy = cast<Torch::ValueTensorType>(imageShape.getType());
+        auto imageShapeSizes = imageShapeTy.getSizes();
+
+        auto blockShapeTy = cast<Torch::ValueTensorType>(blockShape.getType());
+        auto blockShapeSizes = blockShapeTy.getSizes();
+
+        // TODO: Add support for 5D input tensors.
+        if (imageShapeSizes[0] != 2)
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected length of imageShape to be equal to 2");
+        if (blockShapeSizes[0] != 2)
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected length of blockShape to be equal to 2");
+
+        if (dilations.size() != 2)
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected length of dilations to be equal to 2");
+
+        if (strides.size() != 2)
+          return rewriter.notifyMatchFailure(
+              binder.op, "Expected length of strides to be equal to 2");
+
+        // TODO: Disable this check and add support for different
+        // paddings on lower and higher ends of each axis.
+        // Because we have already checked that imageShape has 2 elements,
+        // we can safely assume that len(padding) will be 4.
+        if (pads[0] != pads[2] || pads[1] != pads[3])
+          return rewriter.notifyMatchFailure(
+              binder.op, "padding on the lower end and the higher end "
+                         "on each axis should be the same");
+
+        // Since we know that the padding on the lower end and the higher
+        // end on each axis is the same, we can reduce the size of the
+        // padding list, and filter out the duplicate elements.
+        // (Also, Torch::AtenCol2imOp requires len(padding) to be 2).
+        SmallVector<int64_t> padOnEachAxis = {pads[0], pads[1]};
+
+        SmallVector<Value> cstDilations, cstPadding, cstStrides;
+        for (int64_t i : dilations) {
+          Value x = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(i));
+          cstDilations.push_back(x);
+        }
+        for (int64_t i : padOnEachAxis) {
+          Value x = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(i));
+          cstPadding.push_back(x);
+        }
+        for (int64_t i : strides) {
+          Value x = rewriter.create<Torch::ConstantIntOp>(
+              binder.getLoc(), rewriter.getI64IntegerAttr(i));
+          cstStrides.push_back(x);
+        }
```

Comment:
You can do this in 1 single loop since all will run for same number of iterations.

Reply:
Completely removed by the use of `createConstantIntList()`.